### ISSUE TITLE
[RFE] Support alternative datastores when deploying K3s clusters

### ISSUE THREAD

**Issue by DillonN** [rancher/rancher#44911]
**Is your feature request related to a problem? Please describe.**
Currently when deploying a K3s cluster via a node provisioner such as Harvester, etcd is a required component. There is no way to deploy a cluster that uses an alternative datastore such as MySQL or PostgreSQL

**Describe the solution you'd like**
At a minimum it should be possible to deploy a cluster without any etcd nodes when using K3s. For a full experience, it would be best to have UI options on K3s clusters to select a datastore without having to manually set command line arguments.

**Describe alternatives you've considered**
It's probably possible to deploy a node pool with control and etcd both enabled, then just swap the datastore after creation. However this seems wasteful as etcd would still be running.

And of course it's possible to create the cluster outside of Rancher and then import it afterward. But this is losing some of the value add that Rancher has for deployment.

**Additional context**
Not sure if there'd be a way to work this into the etcd snapshot feature, or if that would be incompatible with clusters created this way.

K3s article on datastores: https://docs.k3s.io/datastore

**Comment by DillonN** [comment]
Still relevant to me

**Comment by dkulchinsky** [comment]
we would love to see native support for this as well 🙏🏼

**Comment by DillonN** [comment]
Still relevant

**Comment by mends-open** [comment]
Still needed

**Comment by chulucninh09** [comment]
Still needed

**Comment by chulucninh09** [comment]
still needed, please don't stale

**Comment by srayancha** [comment]
Yes, we'd love this as well.  We deploy lots of single node clusters that need to work via DHCP. Right now we have to manually install K3s and then import if we go that route. We'd prefer to make these custom clusters and just specify SQLite as the datastore.

**Comment by DillonN** [comment]
This is still relevant
